### PR TITLE
moveit2: fix build for Jazzy

### PR DESCRIPTION
**Pull request by okvik**
This fixes the build of MoveIt2 for Jazzy. The patches likely apply equally for Rolling but I have no time to test this currently.

It's been a while since I did this so exact details escape me, and again, I lack time to re-research, but in short:

1) Several packages have constrained their version requirement on `liboctomap` to <1.10 to fix some distro specific issues on Ubuntu, while what we have is =1.10. As far as I understand the issue is irrelevant on Nix. The fix is to relax the version requirement.
2) `osqp_vendor` required a bit of convincing to get it to build and install like we want.

### CODE REVIEW

File: distros/jazzy/overrides.nix [hunk 1]
```diff
@@ -184,6 +192,51 @@ in {
...
+    postPatch = postPatch + ''
+      substituteInPlace CMakeLists.txt --replace-fail 'find_package(octomap 1.9.7...<1.10.0 REQUIRED)' 'find_package(octomap REQUIRED)'
+    '';
+  });
+
+  osqp-vendor = lib.pipe rosSuper.osqp-vendor [
+    # Make CMakeLists.txt amenable to automatic patching by the next step.
+    (pkg: pkg.overrideAttrs ({ prePatch ? "", ... }: {
+      prePatch = prePatch + ''
+        sed -i 's/GIT_TAG.*/GIT_TAG v0.6.2/' CMakeLists.txt
```
Comment (wentasah) [inline distros/jazzy/overrides.nix:215]: What about being more explicit? Something like this:
```suggestion
        substituteInPlace CMakeLists.txt --replace-fail 'set(git_tag "v0.6.2")' 'set(git_tag "v0.6.2")' # fail when upstream version changes
        substituteInPlace CMakeLists.txt --replace-fail 'GIT_TAG ''${git_tag}' 'GIT_TAG v0.6.2'
```

**Review by wentasah — CHANGES_REQUESTED** [review]
The same changes seem to be needed for rolling. What about placing it in `distro-overlay.nix` so that it can be applied for both versions, similarly as in https://github.com/lopsided98/nix-ros-overlay/issues/511#issuecomment-2425252050.

File: distros/jazzy/overrides.nix (hunk 1 above)
Comment (okvik) [reply distros/jazzy/overrides.nix:215]: Sounds good.

**Comment by okvik** [comment]
> The same changes seem to be needed for rolling. What about placing it in `distro-overlay.nix` so that it can be applied for both versions, similarly as in [#511 (comment)](https://github.com/lopsided98/nix-ros-overlay/issues/511#issuecomment-2425252050).

It surely applies to Rolling, at least currently. It's just unclear to me what we want to happen to Rolling (and future distros) when the patches inevitably fail to apply (due to updates to versions); also what versions and other things might be needed for older distros. None of this I had time or interest in checking because I don't use these, so it just didn't seem fair of me to assume. But looking at it from another point of view, we can let things break and fix when they actually come up.

**Pull request merged**